### PR TITLE
LRDOCS-5743 chart taglibs update

### PR DESCRIPTION
**Pull request by mwilliams2014**
@jbalsas here are the updates to the chart taglib tutorials. It adds steps and additional information that is required if you want to use these taglibs in your portlets. Please review and let me know if anything seems incorrect or out of place. Thanks!

### CODE REVIEW

File: develop/tutorials/articles/210-front-end-taglibs/05-chart-taglibs/12-refreshing-charts-to-reflect-real-time-data.markdown [hunk 1]
```diff
@@ -2,75 +2,73 @@
...
-    							id: 'data2'
-    						}
-    					]
-    				);
-    			};
-    		}
-    	);
-    </aui:script>
+
+2.  To feed real time data to your chart, you can use the chart's component ID
```
Comment (jbalsas) [inline develop/tutorials/articles/210-front-end-taglibs/05-chart-taglibs/12-refreshing-charts-to-reflect-real-time-data.markdown]: This is actually just demo code and one of the options. The preferred way would be to simply use the `data` attribute as an endpoint that changes over time. 

I'm not sure how hard would it be for you to setup a service, but the idea is that you point to something like `https://my.dynamic/stock/exchange` url that gives you updated information about, for example stock exchange. Anytime you poll it, you get the new data.

The case you're showing is when you're using the chart component from javascript, and if a function is configured, then the method is called on every polling interval. The result, in the demo, is that random data points are returned. 

Since we're showing the tag usage, I think it's better if we could focus on the first aspect of the API. We don't need to build the service for it, just show the same screencast and mention that our service is returning real time data that changes overtime and that's it!

File: develop/tutorials/articles/210-front-end-taglibs/05-chart-taglibs/11-predictive-charts.markdown [hunk 2]
```diff
@@ -1,146 +1,215 @@
...
+10.  Add the `<chart>` taglib to the `view.jsp`. An example configuration is 
+     shown below:
+
+        <chart:predictive
+         componentId="predictive-chart"
+         config="<%= predictiveChartSampleDisplayContext.getPredictiveChartConfig() %>"
+         id="predictive-chart"
+        />
+
+11.  Finally, this example adds data via an `aui:script`. It references the
```
Comment (jbalsas) [inline develop/tutorials/articles/210-front-end-taglibs/05-chart-taglibs/11-predictive-charts.markdown]: What's the goal of showing this for the predictive chart in particular? And how is it tied to specifically "adding data after the chart is loaded" in this case?

Seeing this and the other example, I think it might be good to have something like "Programatically Using Chart Components" or something along those lines... where we explain how to grab the chart instance (`Liferay.componentReady`) and then simply outline and describe the available APIs. You can reach out to julien and matu to get you a draft of those :)

Comment (mwilliams2014) [reply develop/tutorials/articles/210-front-end-taglibs/05-chart-taglibs/11-predictive-charts.markdown]: You're right. This is extra information that could be included in a separate tutorial that you mentioned. I'll reach out to them for the draft.

File: develop/tutorials/articles/210-front-end-taglibs/05-chart-taglibs/12-refreshing-charts-to-reflect-real-time-data.markdown (hunk 1 above)
Comment (mwilliams2014) [reply develop/tutorials/articles/210-front-end-taglibs/05-chart-taglibs/12-refreshing-charts-to-reflect-real-time-data.markdown]: That makes sense. A practical example that shows how to use the data attribute as an endpoint would be useful then.

File: develop/tutorials/articles/210-front-end-taglibs/05-chart-taglibs/01-bar-charts.markdown [hunk 3]
```diff
@@ -1,40 +1,106 @@
...
-    import com.liferay.frontend.taglib.chart.model.point.bar.BarChartConfig;
-    
-    public class ChartSampleDisplayContext {
-  
-        public ChartSampleDisplayContext() {
-        	_initBarChartConfig();
-        }
+1.  [Create an MVC portlet project](/develop/reference/-/knowledge_base/7-1/using-the-mvc-portlet-template).
+
+2.  Create an additional *ChartDisplayContext* class and add the following
```
[the remark this answers is not in the document]
Comment (mwilliams2014) [reply develop/tutorials/articles/210-front-end-taglibs/05-chart-taglibs/01-bar-charts.markdown]: I've heard over the years the overcrowding the JSP with java scriptlets is not really best practice though. It seems more like a hack then demonstrating the best practice approach to using charts in your portlets. 

What do you consider to be the best practice approach? If it's using the display context to separate the code, as the examples in portal demonstrate, then that's the approach we should document.

Thoughts @jbalsas ?

File: develop/tutorials/articles/210-front-end-taglibs/05-chart-taglibs/11-predictive-charts.markdown (hunk 2 above)
Comment (mwilliams2014) [reply develop/tutorials/articles/210-front-end-taglibs/05-chart-taglibs/11-predictive-charts.markdown]: Ok. I emailed both of them already 😄

Comment (mwilliams2014) [reply develop/tutorials/articles/210-front-end-taglibs/05-chart-taglibs/11-predictive-charts.markdown]: Alright, taken care of 👍

File: develop/tutorials/articles/210-front-end-taglibs/05-chart-taglibs/01-bar-charts.markdown (hunk 3 above)
[the remark this answers is not in the document]
Comment (jbalsas) [reply develop/tutorials/articles/210-front-end-taglibs/05-chart-taglibs/01-bar-charts.markdown]: Sure, Java scriptlets is not pretty, but it's not a bad practice nor even a hack. It's something that the language offers.

I don't care about "best practices", nor should you in this case. Our goal here is:
- Show how easy it is to use the charts tag
- Show what options are available to you
- Show what's the result of using a chart tag

By showing what someone (we?) consider best practice, we're completely losing 1. The tutorial gets overly complicated, hard to follow and gives the impression that using charts is nightmarish :)

On the other hand, developers might be already following other "best practices" not dictated by us, but by their companies or their workflows. For example, they might be using shared static utils, or injected services, or common tags... it's not our goal in these tutorials to present any best practice.

As a matter of fact, the simplest you can do is point the `data` to attribute to a `json` file and just show that json file. Then, you have the programatic Java API for when you need to generate that output. If you don't want to show a java scriplet (which is totally fine), just do something like: `In a different file, in a galaxy far far away...` and simply show the snippet of Java code. A developer would know how to take that and put it in a jsp or in any place they have to to get it working.

**Comment by jbalsas** [comment]
Closing here, as I expect some changes :)

Feel free to continue over this discussion or ping me if you have any doubts!

File: develop/tutorials/articles/210-front-end-taglibs/05-chart-taglibs/01-bar-charts.markdown (hunk 3 above)
[the remark this answers is not in the document]
Comment (mwilliams2014) [reply develop/tutorials/articles/210-front-end-taglibs/05-chart-taglibs/01-bar-charts.markdown]: As long as we cover all the requirements so the developers are not left scratching their head trying to fill in the missing pieces, I'm happy 😄

[the remark this answers is not in the document]
Comment (jbalsas) [reply develop/tutorials/articles/210-front-end-taglibs/05-chart-taglibs/01-bar-charts.markdown]: You'd just need to add the necessary imports to the jsp, which are... 2? 😂 

Honestly, they scratch their heads now trying to figure out why doing simple tasks is so complex. That's real feedback I get all the time about our docs.

Quoting Steve Balmer... "Simplicity! Simplicity! Simplicity!" (I think I might be misremembering this one ;) )

[the remark this answers is not in the document]
Comment (mwilliams2014) [reply develop/tutorials/articles/210-front-end-taglibs/05-chart-taglibs/01-bar-charts.markdown]: Noted
